app/javascript/packs/components/post_form.tsx

```diff
@@ -88,6 +92,30 @@ const PostForm: React.FC<PostProps> = (props) => {
               />
             </div>
           </div>
+          {/* Only to logged in user and if that user belongs to group 
+          we will show this option*/}
+          <div className="md:flex md:items-center mb-6">
+            <div className="md:w-1/3">
+              <div className="md:w-1/3">
+                <label className="block text-black md:text-xl sm:text-xl font-bold md:text-right mb-1 md:mb-0 pr-4">
+                  Send to Group
+                  <input type="checkbox" 
+                         name="post[group_selected]" 
+                         onChange={() => setGroupChecked(!groupChecked)}
+                          checked={groupChecked}
+                         id=""/>
+                  <br/>
+                  {
+                  groupChecked ? (
+                    <input className="appearance-none border-2 border-gray-200 rounded w-full py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-middle-blue-green"
+                            name="post[group_name]" type="text" 
+                            placeholder="Enter group name"/>
+                    ) : (<div></div>)
+                  }
+                </label>
+              </div>
+            </div>
+          </div>
```

Comment:
@saurabhbhatia  Mamu, this is where I need help. So the idea is, if the user is logged in we show group checkbox and on click, we show and hide the group field which will be autocompleted. created_by will be current_user id. Below is the payload we need to pass to the post create API.
{

"body": "hello",
"salty_password": "1",
"has_salt": true,
"expired_at": 24,
"group_selected": true,
"group_id": 3,
"created_by": 5

}